### PR TITLE
Added location options in CREATE/EDIT User AND move logged in user.

### PR DESCRIPTION
**Pull request by MahadJanjua**
-Location options in Create User
-Location options in Edit User
-Location options in Current User
-Should all include "Outside" functionality despite it not being a room (outside is treated as `user.setLocation(null)`)

### CODE REVIEW

**Comment by adrientremblay** [comment]
closes #24

File: src/Controller/DashboardController.java [hunk 1]
```diff
@@ -149,17 +198,37 @@ else if (editUserType.getValue() == null && editUserNewPassword.getText().length
...
+		}
+
+		// Change location if that had input
+		if (editUserLocation.getValue().equals("Outside")) {
+			sim.getLoggedInUser().setLocation(null);
+		}
+		else if (editUserLocation.getValue() != null) {
+			for (Room r : sim.getHouse().getRooms()) {
+				if (r.getName().equals(editUserLocation.getValue())) {
+					sim.getLoggedInUser().setLocation(r);
```
Comment (adrientremblay) [inline src/Controller/DashboardController.java]: I feel like there should be a check to see if the new location is the same as the old location.  But maybe this check should be done within the setLocation method within the User class.  Maybe we should make the printToConsole method public so that errors can be printed from inside Model classes such as User and Simulation.

File: src/Controller/DashboardController.java [hunk 2]
```diff
@@ -20,34 +20,67 @@
 
 public class DashboardController {
 	private Simulation sim;
+	private int windowLength = 30;
+	private int ROOM_SIZE = 50;
 
 
 	@FXML private Label currentUser;
+	@FXML private Label currentUserLocation;
+	@FXML private ComboBox<String> currentUserLocationOptions;
```
Comment (adrientremblay) [inline src/Controller/DashboardController.java:28]: Theres a problem with consistency when theres both a current user label and a combo box because these values can differ.  Maybe we should scrap the label.  Just have the combobox represent the current user's position.  But we'd have to write code to update the combobox when someone edits the user.

File: src/Controller/DashboardController.java [hunk 3]
```diff
@@ -56,10 +89,12 @@
...
-		printToConsole("ERROR: Did not find any user with the entered name:" + loginName.getText() + ".");
+		printToConsole("ERROR: Did not find any user with the entered name (" + loginName.getText() + ") and password.");
+		loginName.setText("");
+		loginPassword.setText("");
 	}
 
-	@FXML private void addUser() {
+	@FXML private void createUser() {
 		UserType type = null;
 		if (createUserName.getText().length() == 0) {
```
Comment (adrientremblay) [inline src/Controller/DashboardController.java]: as I said in another comment.  Maybe we should refactor to move this logic to the the Simulation class.  This is low priority though.

File: src/Controller/DashboardController.java [hunk 4]
```diff
@@ -201,6 +269,12 @@ private void printToConsole(String output) {
 	private void updateDashboard() {
 		// reset name of logged in user
 		currentUser.setText(sim.getLoggedInUser().getName());
+		if (sim.getLoggedInUser().getLocation() == null) {
```
Comment (adrientremblay) [inline src/Controller/DashboardController.java:95]: When I test 'Current Location: ...' never changes for me.

Comment (adrientremblay) [reply src/Controller/DashboardController.java:95]: i realized this is because the label is too narrow.  It should be given a bigger minWidth or something.  So this is executing but we can never see the result.

File: src/View/Dashboard.fxml [hunk 5]
```diff
@@ -18,6 +18,7 @@
             <Text strokeType="OUTSIDE" strokeWidth="0.0" text="Simulation:"/>
             <ToggleButton mnemonicParsing="false" prefHeight="27.0" prefWidth="141.0" text="ON/OFF"/>
             <Label fx:id="currentUser" text="Username (or login)"/>
+            <Label fx:id="currentUserLocation" text="Current Location: "/>
```
Comment (adrientremblay) [inline src/View/Dashboard.fxml]: Should be given an appropriate width property (see comment above)

**Review by adrientremblay — CHANGES_REQUESTED** [review]

File: src/Controller/DashboardController.java (hunk 1 above)
Comment (MahadJanjua) [reply src/Controller/DashboardController.java]: Hm I think you're right for moving the logic to the Simulation class. However, we definitely shouldn't move printToConsole. Model should include all the logic to handle the simulation, controller should be using the functions in model to handle the situations it's provided, but printing to the console doesn't have anything to do with the house system/model, it's only a method to give back alerts and messages on task completions and stuff.

TL:DR; let's resolve the PRs for now and I will see if it's possible to move the logic of user creation/editing/deletion into the simulation class as it fits better in the MVC model.

File: src/Controller/DashboardController.java (hunk 2 above)
Comment (MahadJanjua) [reply src/Controller/DashboardController.java:28]: Yeah the updateDashboard function just needs to update all the values to be blank but I couldn't figure out how to do that (mostly because I kind of ignored it for the time being). Will check how to clear/reset combobox values.

**Review by adrientremblay — APPROVED** [review]
